### ISSUE TITLE
No error for duplicate unpacking variable

### ISSUE THREAD

**Issue by aure-olli** [PyCQA/pylint#2898]
### Is your feature request related to a problem? Please describe
```python
first, first = (1, 2)
print(first, first)

for second, second in enumerate(range(5, 10)):
        print(second, second)
```
This code doesn't produce any error related to the fact that twice the same variable is used when unpacking. I don't see any case where it would not be a bug (except when using `_` variable).

### Describe the solution you'd like

    test.py:1:0: CXXXX: duplicate variable "first" when upacking (duplicate-unpacking-variable)
    test.py:4:4: CXXXX: duplicate variable "second" when upacking (duplicate-unpacking-variable)

### Additional context

I was quite surprised this isn't already checked. I hope this is possible.

**Comment by PCManticore** [comment]
It's definitely possible to have a check for this, thanks for reporting the issue!
